### DISCUSSION TITLE
Tabview: How to activate disabled Tab?

### DISCUSSION THREAD

**Discussion by hgfrey** [primefaces/community#2865] [PrimeFaces] [answered]
Since version 14.0.0, a subsequently activated tab (not longer disabled) is no longer displayed as selected after the tab has been selected. This still worked in version 13.0.10.

If the tab is active from the start, the layout also works; if the tab is selected, it is displayed that way.

**Comment by melloware** [comment]
Can you provide a PF Test reproducer showing the issue?  Are you using 14.0.5?

**Reply by hgfrey** [reply]
Yes, we are using 14.0.5.

I will assemble an example. But that will take a day or two because there are more important things to do. ;-)

**Comment by hgfrey** [comment]
Which element do I have to update so that the tab is changed from inactive to active and displayed like that? I can't use @form because then all the previously entered entries are gone.

[primefaces-test-master.zip](https://github.com/user-attachments/files/17465102/primefaces-test-master.zip)

**Reply by melloware** [reply]
Before i investigate did this ever work before?  like does this reproducer work on 13.0.0 and not on 14.0.6?

**Reply by hgfrey** [reply]
Not in this case I sent. In fact, I'm wondering why this actually worked in our application up to version 14.0.0. I couldn't find anything so far. Maybe I have to search again.

**Reply by melloware** [reply]
I am guessing it did NOT ever work so its not  a bug or new bug

**Reply by melloware** [reply]
I edited your title "Tabview: How to activate disabled Tab?"

**Reply by hgfrey** [reply]
But actually I would have expected that this wouldn't work without a suitable update.

**Reply by hgfrey** [reply]
> I am guessing it did NOT ever work so its not a bug or new bug

I think you are right.

**Reply by hgfrey** [reply]
> I edited your title "Tabview: How to activate disabled Tab?"

I don't understand. What should I do?

**Reply by melloware** [reply, marked as answer]
See this ancient post from 2012: https://forum.primefaces.org/viewtopic.php?t=17754  asking the same question you are asking.  TabView is the container as individual tabs are not components.

**Reply by hgfrey** [reply]
If I do it like this, the previously entered content is gone.

**Reply by hgfrey** [reply]
[primefaces-test-master.zip](https://github.com/user-attachments/files/17465369/primefaces-test-master.zip)

**Reply by melloware** [reply]
yep i am just telling you the TabView is the component and not tabs so individual tabs cannot be updated.

**Reply by hgfrey** [reply]
OK. Unfortunately lost...

**Reply by melloware** [reply]
Don't use TabView use TabMenu and treat all your forms as separate pages if you need this behavior: https://www.primefaces.org/showcase/ui/menu/tabMenu.xhtml

**Comment by hgfrey** [comment]
Understood. Thanks for the suggestion. I will discuss this with my colleagues and the product owner.
